app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/LocalFileTransferActivity.java

```diff
@@ -0,0 +1,398 @@
+package org.kiwix.kiwixmobile.zim_manager.local_file_transfer;
+
+import android.Manifest;
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.location.LocationManager;
+import android.net.Uri;
+import android.net.wifi.p2p.WifiP2pDevice;
+import android.net.wifi.p2p.WifiP2pDeviceList;
+import android.os.Build;
+import android.os.Bundle;
+import android.provider.Settings;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.ListView;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+import android.widget.Toast;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.widget.Toolbar;
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnItemClick;
+import java.util.ArrayList;
+import java.util.List;
+import javax.inject.Inject;
+import kotlin.Unit;
+import kotlin.jvm.functions.Function0;
+import org.kiwix.kiwixmobile.KiwixApplication;
+import org.kiwix.kiwixmobile.R;
+import org.kiwix.kiwixmobile.utils.AlertDialogShower;
+import org.kiwix.kiwixmobile.utils.KiwixDialog;
+import org.kiwix.kiwixmobile.utils.SharedPreferenceUtil;
+
+import static org.kiwix.kiwixmobile.zim_manager.local_file_transfer.WifiDirectManager.getDeviceStatus;
+
+/**
+ * Created by @Aditya-Sood as a part of GSoC 2019.
+ *
+ * This activity is the starting point for the module used for sharing zims between devices.
+ *
+ * The module is used for transferring ZIM files from one device to another, from within the
+ * app. Two devices are connected to each other using WiFi Direct, followed by file transfer.
+ *
+ * File transfer involves two phases:
+ * 1) Handshake with the selected peer device, using {@link PeerGroupHandshakeAsyncTask}
+ * 2) After handshake, starting the files transfer using {@link SenderDeviceAsyncTask} on the sender
+ * device and {@link ReceiverDeviceAsyncTask} files receiving device
+ */
+@SuppressLint("GoogleAppIndexingApiWarning")
+public class LocalFileTransferActivity extends AppCompatActivity implements
+  WifiDirectManager.Callbacks {
+
+  // Not a typo, 'Log' tags have a length upper limit of 25 characters
+  public static final String TAG = "LocalFileTransferActvty";
```

Comment:
How about starting the log message with kiwix to make the log messages easier for kiwix developers to detect and filter? e.g. kiwixFileTransfer or kiwixLocalFileTransfer?

Reply:
I think they can already be filtered on the basis of the app creating them? Like android studio has an option in logcat to filter the logs being displayed based on which debuggable app has been selected in the filter

Reply:
A) Yes it does but julian is talking about working with raw log files? grepping application Id should work in that scenario?
B) we are going to only log on debug #1270 soon and with our custom logger, it can handle TAG prefixing or any pre/post processing we need to do on logs (writing logs to a file ourselves, truncating tags, null checking messages so the native call doesn't crash)
C) this PR has been merged so requesting changes should be done via a ticket probably?